net/proxy_resolution/proxy_resolution_service_unittest.cc

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "net/proxy_resolution/proxy_resolution_service.h"
+
+#include <string>
+
+#include "base/macros.h"
+#include "base/memory/ptr_util.h"
+#include "brave/net/proxy_resolution/proxy_config_service_tor.h"
+#include "net/base/test_completion_callback.h"
+#include "net/log/test_net_log.h"
+#include "net/proxy_resolution/mock_proxy_resolver.h"
+#include "net/test/gtest_util.h"
+#include "net/test/test_with_scoped_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using net::test::IsOk;
+
+namespace net {
+
+class ProxyResolutionServiceTest : public TestWithScopedTaskEnvironment {
+ public:
+  ProxyResolutionServiceTest() {}
+  ~ProxyResolutionServiceTest() override {}
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(ProxyResolutionServiceTest);
+};
+
+TEST_F(ProxyResolutionServiceTest, TorProxy) {
+  MockAsyncProxyResolverFactory* factory =
```

Comment:
nits: `make_unique` here and `std::move` below
const everywhere for local variables that do not change (`proxy_uri`, `site_url`, etc)

Reply:
MockAsyncProxyResolverFactory will be used later

Reply:
we don't actually care about `factory->pending_requests().empty` so I will wrap it when constructing `ProxyResolutionService`

Reply:
fixed in 58622cb17913fb4e4925f1d2943f2d561a4bdbd4

Reply:
fixed in 58622cb17913fb4e4925f1d2943f2d561a4bdbd4